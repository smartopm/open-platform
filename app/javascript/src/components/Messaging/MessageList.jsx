import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import MaterialList from '@material-ui/core/List'
import UserMessageItem from './UserMessageItem'
import Nav from '../Nav'

export default function MessageList({ messages }) {
  return (
    <Fragment>
      <Nav navName="Messages" menuButton="back" />
      <MaterialList>
        {messages.length ? (
          messages.map(message => (
            <UserMessageItem
              key={message.user.id}
              id={message.user.id}
              name={message.user.name}
              user={message.user}
              message={message.message}
              clientNumber={message.user.phoneNumber}
              dateMessageCreated={message.createdAt}
              isTruncate={true}
            />
          ))
        ) : (
          <div>
            <p className="text-center">No messages</p>
          </div>
        )}
      </MaterialList>
    </Fragment>
  )
}
MessageList.defaultProps = {
  messages: []
}

MessageList.propTypes = {
  messages: PropTypes.array.isRequired
}
