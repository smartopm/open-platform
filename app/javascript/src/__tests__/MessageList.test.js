import React from 'react'
import { mount, shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom'
import MessageList from '../components/Messaging/MessageList'

describe('message list component', () => {
  const messages = [
    {
      message: 'Heyyyy there',
      isRead: true,
      dateMessageCreated: '2020-04-13',
      readAt: '2020-04-14',
      id: '1124ff7d-0008-02b4c5a54b50',
      user: {
        id: 'b5ed9ea9-8b02-eb8bb4fed2c8',
        name: 'Doez JM'
      }
    }
  ]

  const messagesList = shallow(
    <BrowserRouter>
      <MessageList messages={messages} />
    </BrowserRouter>
  )
  it('message list component should render just fine', () => {
    expect(messagesList).toMatchSnapshot()
  })
  it('shouldn\' show no message text when messages are provided', () => {
    expect(messagesList.find('.nz_no_msg')).toHaveLength(0)
  })
  it('should show no messages when given an empty array', () => {
    const messages = []
    const EmptyMessagesList = mount(<MessageList messages={messages} />)
    expect(EmptyMessagesList.find('.nz_no_msg').text()).toContain('No messages')
  })
})
