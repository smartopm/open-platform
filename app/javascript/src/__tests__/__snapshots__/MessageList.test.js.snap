// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message list component message list component should render just fine 1`] = `ShallowWrapper {}`;
